src/integration-test/integration/ODataClient.test.ts

```diff
@@ -0,0 +1,121 @@
+import "reflect-metadata";
+import { accessToken, projectId, sleep, testIModel, testIModelGroup } from "../utils";
+use(chaiAsPromised);
+
+describe("OData Client", () => {
+  const oDataClient: ODataClient = new ODataClient();
+  const reportsClient: ReportsClient = new ReportsClient();
+  const mappingsClient: MappingsClient = new MappingsClient();
+  const extractionClient: ExtractionClient = new ExtractionClient();
+
+  let reportId: string;
+  let oDataItem: ODataItem;
+  const deletionTracker: Array<string> = [];
+
+  before(async function () {
+    const newMapping: MappingCreate = {
+      mappingName: "Test",
+    };
+    const mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test");
+    deletionTracker.push(mapping.id);
+
+    const newGroup: GroupCreate = {
+      groupName: "Test",
+      query: "select * from biscore.element limit 10",
+    };
+    const group = await mappingsClient.createGroup(accessToken, testIModel.id, mapping.id, newGroup);
+    expect(group).to.not.be.undefined;
+    expect(group.groupName).to.be.eq("Test");
+
+    const newReport: ReportCreate = {
+      displayName: "Test",
+      projectId,
+    };
+    const report = await reportsClient.createReport(accessToken, newReport);
+    expect(report).to.not.be.undefined;
+    expect(report.displayName).to.be.eq("Test");
+    deletionTracker.push(report.id);
+    reportId = report.id;
+
+    const newReportMapping: ReportMappingCreate = {
+      mappingId: mapping.id,
+      imodelId: testIModel.id,
+    };
+    const reportMapping = await reportsClient.createReportMapping(accessToken, report.id, newReportMapping);
+    expect(reportMapping).to.not.be.undefined;
+    expect(reportMapping.mappingId).to.be.eq(mapping.id);
+    deletionTracker.push(reportMapping.mappingId);
+
+    const extraction = await extractionClient.runExtraction(accessToken, testIModel.id);
+    expect(extraction).to.not.be.undefined;
+
+    let state = ExtractorState.Queued;
+    let status: ExtractionStatus;
+    for (const start = performance.now(); performance.now() - start < 6 * 60 * 1000; await sleep(3000)) {
+      status = await extractionClient.getExtractionStatus(accessToken, extraction.id);
+      state = status.state;
+      if(state !== ExtractorState.Queued && state.valueOf() !== ExtractorState.Running)
+        break;
+    }
+    expect(state).to.be.eq(ExtractorState.Succeeded);
+
+    const oDataResponse = await oDataClient.getODataReport(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse.value).to.not.be.empty;
+    oDataItem = oDataResponse.value[0];
+
+  });
+
+  after(async function () {
+    let i = 0;
+    if(i < deletionTracker.length) {
```

Comment:
This kind of cleanup is very error prone. Simply add a couple more variables and store each id separately.

Reply:
Exchanged for a simple cleanup